elastic/kibana
# [Security Solution]  Make Timeline a modal

## Summary

Issue: https://github.com/elastic/kibana/issues/169021

Changes the Timeline layout to look like a modal.
These are only visual changes, all existing behaviors in the Timeline will stay the same.

The Timeline styles are custom, `EuiOverlayMask` styles have been applied to the wrapper, and `margin` to the timeline content, to have a modal-like look and feel, as defined.

Since the Timeline is always rendered and just "hidden" when closed, using `EuiModal` or `EuiOverlayMask` directly resulted in a more complex implementation for different reasons, especially because they set `overflow: hidden` globally to the document `body` when rendered. This PR does the same thing but only when the timeline is "visible"

### Test

1. Open the Security app (Serverless or ESS)
2. Open Timeline, it should look as in the design specification.
3. All the existing behaviors in the Timeline should keep working the same way.

## Screenshots

Serverless:

<img width="1717" alt="Captura de pantalla 2023-11-08 a les 13 10 22" src="https://github.com/elastic/kibana/assets/17747913/a6667c29-f2df-43fc-90f1-716c74468d71">


ESS:

<img width="1717" alt="Captura de pantalla 2023-11-08 a les 13 15 06" src="https://github.com/elastic/kibana/assets/17747913/e6973cca-111f-4715-b053-732207c5f399">
<img width="1717" alt="Captura de pantalla 2023-11-08 a les 13 15 18" src="https://github.com/elastic/kibana/assets/17747913/a2956475-cca2-43a8-8f83-6672ad68582b">
